extensions/interactions/NumericExpressionInput/directives/numeric-expression-input-interaction.component.html

```diff
@@ -3,3 +3,31 @@
 <div ng-show="!$ctrl.isCurrentAnswerValid()" class="alert alert-danger mt-2">
   <[$ctrl.warningText]>
 </div>
+<!--
+  Guppy enforces the 'white-space', 'height' and 'width' styles causing the
```

Comment:
This doesn't look OK to me. It's fine for .guppy-div-learner to be here, but the other classes shouldn't be. Why can't they be in extensions/interactions/guppy.html? Also those classes are very broad and the CSS selector needs to be much more specific so that they are contained only to the component they are supposed to affect.

The comment provided is also a bit off, because it sounds like "the stuff below breaks our project design principles but we'll do it anyway". Instead, you'll want to explain, above the specific lines in question: "This is what Guppy does. We have tried specific CSS selectors but this does not work because XXX. Therefore, !important is needed."  XXX should be a technical reason (not just "we tried it and it didn't work") and the whole comment should constitute a complete "proof", that is as rigorous as possible, for why we need to break convention here.

Reply:
I tried other ways to fix this, but I couldn't come up with a better (or rather, more intuitive) solution. The reason is that Guppy seems to override the styles given anywhere with its own predefined styles. The only way to fix it is to override the specificity with !important (since Guppy itself seems to be doing so). 

I have improved the comments a bit to explain this so that it becomes more maintainable.